src/halmos/sevm.py

```diff
@@ -1654,7 +1655,14 @@ def balance_of(self, addr: Word) -> Word:
         self.path.append(Select(EMPTY_BALANCE, addr) == ZERO)
 
         # practical assumption on the max balance per account
-        self.path.append(ULT(value, con(2**96)))
+        # (only for symbolic values, otherwise we may append false constraints)
+        if is_bv_value(value):
+            if (v := value.as_long()) > MAX_ETH:
+                warn(
```

Comment:
after sleeping on it, I wonder if we should raise an exception here instead of printing a warning. Otherwise it's a bit strange to have a constraint like `ULE(value, con(MAX_ETH))`, while some values may in fact be higher than MAX_ETH

Reply:
yeah we wouldn't want to continue once the value becomes larger than the max value, as any counterexamples wouldn't make sense anyway. so it might be better to raise an exception here, and let the outer loop handle it properly depending on context.